[WiP] Fix untranslated com_installer language strings in com_joomlaupdate when an exception is raised on Upload & Update

richard67 opened the pull request:
Pull Request for [https://github.com/joomla/joomla-cms/pull/29884#issuecomment-652615993](https://github.com/joomla/joomla-cms/pull/29884#issuecomment-652615993) .

### Summary of Changes

This Pull Request (PR) changes the language strings used in the upload function of the Joomla Update model when raising exceptions from com_installer strings (which currently are not translated) to com_joomlaupdate or global strings, depending if they should be same for both the installer and the Joomla Update component (ans possibly other, 3rd party extensions) and adds the necessary strings to the com_joomlaupdate global language files. This fixes the strings not being translated, because the com_installer language is not loaded in the com_joomlaupdate model.

In addition it fixes a missing use of translation (i.e. `JText::_` call) here: https://github.com/joomla/joomla-cms/pull/29899/files#diff-35b54662480daa18b046886a634f00d6L951 .

Finally this PR changes the install model of com_installer so it uses the language strings which have been changed to global strings as mentioned before. The previosly used strings are marked as deprecated in the com_installer language file.

This does not change com_installer language string which are used in the default_upload.php view of com_joomlaupdate  Those strings are translated well because the language is loaded, so I did not see the need to change them.

### Testing Instructions

Will be added soon. As long as this is not done I'll leave this PR in draft status.

### Actual result BEFORE applying this Pull Request

Will be added soon.

### Expected result AFTER applying this Pull Request

Will be added soon.

### Documentation Changes Required

None.

richard67 committed:
Use com_joomlaupdate language strings for exceptions on Upload & Update

brianteeman commented:
Would it not be better to use `$lang->load ` to include the existing language file than to create another set of strings

richard67 commented:
> Would it not be better to use `$lang->load ` to include the existing language file than to create another set of strings

@brianteeman I had asked about this and got feedback from @zero-24 that I should not make the joomla update model depend on the installer's language strings, see e.g. [https://github.com/joomla/joomla-cms/pull/29884#issuecomment-652610815](here https://github.com/joomla/joomla-cms/pull/29884#issuecomment-652610815).

brianteeman commented:
Obviously I disagree but if anything then both components should be using a global string.

richard67 commented:
@HLeithner @zero-24 Please advise regarding previous comments. I'll do whatever will be decided.

richard67 commented:
@brianteeman If you compare the strings used without and with this PR, you will see they are not all the same. This PR here changes some of then which talk about the installer so that they talk about Joomla Update. So at least for those global strings will not make sense.

HLeithner commented:
It should be ok if the updater has it's own strings especially if not every thing is exactly the same.

zero-24 commented:
I agree with @HLeithner you might even what to give a different message in com_joomlaupdate maybe in non en-GB languages. The thing about not depending on com_installer is it should be possible that com_joomlaupdate can be released as standalon so it should be done with as less as possible dependencies to other components in my opinion.

richard67 commented:
> I agree with @HLeithner you might even what to give a different message in com_joomlaupdate maybe in non en-GB languages. The thing about not depending on com_installer is it should be possible that com_joomlaupdate can be released as standalon so it should be done with as less as possible dependencies to other components in my opinion.

@zero-24 This PR only solves it for the model, where the installer language strings were not translated. But there are still installer language strings used in the default_upload.php view. Those strings are translated well, so I did not see the need to change them, too. @HLeithner Is that ok, or shall I handle them too with this PR?

HLeithner commented:
I'm unsure... because having duplicated strings are look wrong but using strings from an other component doesn't make sense either....

richard67 committed:
Use global language strings for common texts

richard67 committed:
Move more installer language strings to global

richard67 commented:
@brianteeman @HLeithner I've moved those untranslated strings fixed by this PR and being the same for com_installer and com_joomlaupdate to global language strings and added deprecation comments to those language strings in the com_installer language file. For consistency I've also moved two description texts which belong to other two moved texts, even if those descriptions are (currently) only used by the installer. The installer PHP I've changed to use the moved strings.

@brianteeman I'd be happy if you could check what I've done with the strings, especially check the deprecation comments in the installer language file and if necessary suggest corrections. Thanks in advance.

richard67 commented:
PR is still draft because I have to write testing instructions and I'd like to wait a bit, maybe PR #29881 will be merged soon, and this will make testing of this PR here easier. But it can be already reviewed.

brianteeman commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -24,6 +24,8 @@ COM_JOOMLAUPDATE_MINIMUM_STABILITY_DEV="Development"
 COM_JOOMLAUPDATE_MINIMUM_STABILITY_LABEL="Minimum Stability"
 COM_JOOMLAUPDATE_MINIMUM_STABILITY_RC="Release Candidate"
 COM_JOOMLAUPDATE_MINIMUM_STABILITY_STABLE="Stable"
+COM_JOOMLAUPDATE_MSG_WARNINGS_UPLOADNOTENABLED="Joomla Update can't continue until file uploads are enabled for the server."
```
Not sure what you are trying to say here

HLeithner replied:
It means "File Upload is disabled on the server" it's a php setting

brianteeman replied:
sorry I was being deliberately awkward:) . If you don't understand anything about php how would you know it was not  a joomla setting

richard67 replied:
If I shall change this string to something more meaningful, please make a suggestion and leave a comment if I shall change string `COM_INSTALLER_MSG_INSTALL_WARNINSTALLFILE` in the same way. That string is not touched by this PR here, but it is the one from which the above one was derived from.

brianteeman replied:
thinking more about it - it is so unlikely that this will ever be displayed it can stay as it is. anyone with a server so badly disabled will already know the server sucks and will make the connection.

richard67 replied:
Ok, thanks. Marking as resolved.

brianteeman commented on administrator/language/en-GB/en-GB.com_joomlaupdate.ini:
```diff
@@ -24,6 +24,8 @@ COM_JOOMLAUPDATE_MINIMUM_STABILITY_DEV="Development"
 COM_JOOMLAUPDATE_MINIMUM_STABILITY_RC="Release Candidate"
 COM_JOOMLAUPDATE_MINIMUM_STABILITY_STABLE="Stable"
+COM_JOOMLAUPDATE_MSG_WARNINGS_UPLOADNOTENABLED="Joomla Update can't continue until file uploads are enabled for the server."
+COM_JOOMLAUPDATE_MSG_WARNINGS_ZLIBMISSING="Joomla Update can't continue until Zlib is installed."
```
This needs to be clear that you are referring to something on the server and not an extension from JED ;)

HLeithner replied:
It means "Zlib php extension is missing on the server"

richard67 replied:
The strings are exactly the same which are here since ages for the installer, only that "Installer" has been replaced by "Joomla Update". So if it was wrong, it was wrong all the time, and @brianteeman could have made a PR long time ago.

brianteeman replied:
If I had seen it I would have changed it ;)
there are a lot of language strings I can't be expected to have read them all.

```suggestion
COM_JOOMLAUPDATE_MSG_WARNINGS_ZLIBMISSING="Joomla Update can't continue until Zlib is enabled for the server."
```

richard67 replied:
Shall I change `COM_INSTALLER_MSG_INSTALL_WARNINSTALLZLIB` for the installer in the same way so it's consistent? The string is not touched by this PR, but that's the string where the above one was derived from.

brianteeman replied:
please

richard67 replied:
Ok, will do so. Thanks, marking as resolved as soon as commited.

richard67 replied:
Done.

richard67 commented:
@brianteeman Could you check my replies to your review comments above and make suggestions? Thanks in advance, and thanks for review so far.

brianteeman commented:
@richard67 I already did - or am i missing something

richard67 committed:
Make error message about missing Zlib more clear

richard67 commented:
@brianteeman Thansk for review. All done. Stay tuned with testing instructions and draft status removal. I need a break.

richard67 committed:
Merge remote-tracking branch 'upstream/staging' into staging-fix-untranslated-language-strings-joomlaupdate

richard67 committed:
Merge remote-tracking branch 'upstream/staging' into staging-fix-untranslated-language-strings-joomlaupdate

richard67 committed:
Merge remote-tracking branch 'upstream/staging' into staging-fix-untranslated-language-strings-joomlaupdate

richard67 committed:
Merge remote-tracking branch 'upstream/staging' into staging-fix-untranslated-language-strings-joomlaupdate

richard67 committed:
Merge branch 'staging' into staging-fix-untranslated-language-strings-joomlaupdate

richard67 commented:
It seems the issue with these untranslated strings has been solved meanwhile, since on a current staging they are all translated when I add some debug code to check that.

I've just checked that.
